Content.Shared/Clothing/EntitySystems/EmitsSoundOnMoveSystem.cs

```diff
@@ -0,0 +1,99 @@
+using System.Numerics;
+using Content.Shared.Clothing.Components;
+using Content.Shared.Gravity;
+using Content.Shared.Inventory;
+using Content.Shared.Inventory.Events;
+using Content.Shared.Mobs.Components;
+using Content.Shared.Movement.Components;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Physics.Components;
+using Robust.Shared.Timing;
+
+namespace Content.Shared.Clothing.Systems;
+
+public sealed class EmitsSoundOnMoveSystem : EntitySystem
+{
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedMapSystem _grid = default!;
+    [Dependency] private readonly SharedGravitySystem _gravity = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+
+    private EntityQuery<InputMoverComponent> _moverQuery;
+    private EntityQuery<PhysicsComponent> _physicsQuery;
+    private EntityQuery<TransformComponent> _xformQuery;
+    private EntityQuery<ClothingComponent> _clothingQuery;
+
+    public override void Initialize()
+    {
+        _moverQuery = GetEntityQuery<InputMoverComponent>();
+        _physicsQuery = GetEntityQuery<PhysicsComponent>();
+        _xformQuery = GetEntityQuery<TransformComponent>();
+        _clothingQuery = GetEntityQuery<ClothingComponent>();
+
+        SubscribeLocalEvent<EmitsSoundOnMoveComponent, GotEquippedEvent>(OnEquipped);
+        SubscribeLocalEvent<EmitsSoundOnMoveComponent, GotUnequippedEvent>(OnUnequipped);
+    }
+
+    private void OnEquipped(EntityUid uid, EmitsSoundOnMoveComponent component, GotEquippedEvent args)
+    {
+        component.IsSlotValid = !args.SlotFlags.HasFlag(SlotFlags.POCKET);
+    }
+
+    private void OnUnequipped(EntityUid uid, EmitsSoundOnMoveComponent component, GotUnequippedEvent args)
+    {
+        component.IsSlotValid = true;
+    }
+
+    public override void Update(float frameTime)
+    {
+        var query = EntityQueryEnumerator<EmitsSoundOnMoveComponent>();
+        while (query.MoveNext(out var uid, out var comp))
+        {
+            UpdateSound(uid, comp);
+        }
+        query.Dispose();
+    }
+
+    private void UpdateSound(EntityUid uid, EmitsSoundOnMoveComponent component)
+    {
+        if (!_xformQuery.TryGetComponent(uid, out var xform) ||
+            !_physicsQuery.TryGetComponent(uid, out var physics))
+            return;
+
```

Comment:
xformQuery is deprecated, and the first half of this check is unnecessary now. All systems parented to EntitySystem can instead reference a transform component via 
`Transform(uid)`, which represents the always-available, guaranteed never null, TransformComponent of a given EntityUid.